doc/design/effects_mutability.md

```diff
@@ -32,59 +32,54 @@ made between operating system load versus store effects.
-    `('aa, ^m, >e) 'a^`.
-  - `'a t`, `('a, ^m, >e) t`, ...: Primarily immutable parametric type.
-  - `'a t&`, `('a, ^m, >e) t&`, ...: Primarily mutable parametric type.
-  - `'a& t`: Subsidiarily mutable parametric type.
-  - `'a t ^t`, `'a t^`, `('a, ^m) t^`: Parametric type with parametric primary mutability. `^m` is a
-    subsidiary mutability parameter, whereas the primary mutability parameter `^t` is distinct from
-    the comma-separated `(...)` parameter list, such that its absence and the absence of `&`
-    indicates that the type is primarily immutable.
-  - `val f: 'a -> 'a t^`: The function produces an isolated value with parametric mutability. The
-    value's mutability is either immutable or mutable as determined by type unification in the
+    + `os`: Operating system load-store effect
+    + `ld`: Runtime system load effect
+    + `mut`: Runtime system load-store effect
+    + `hlt`: Runtime system may-halt effect
+  - `val f >e: ... >e-> ...`: Parametric effect.
+  - `val f >e\{os|mut}: ... >e-> ...`: Parametric effects constraint; transitively prohibited effects
+    are enumerated as `\...` following parametric effect.
+  - `(a: type) -> !&t a -> unit`, `(a: type) -> &t !&a >hlt-> unit`: Mutation effect on mutable
+    function parameter type prefixed by `!`. The affected type need not be the outermost type, but
+    the affected type must be mutable.
+  - `(a: type) -> &t a -> &t a`: Load effect on mutable function parameter. A mutable function
+    parameter always implies a `ld` effect on the parameter, without which the parameter would be
+    useless.
+  - `uns >{&t|(!&u)}-> unit`, `!&t >!&t-> unit`: Type-constrained environment `ld`/`mut` effect(s)
+    during function application. Parentheses around type-constrained effects are optional, though
+    they often improve readability. Note that a `!&t` parameter mutation is distinct from
+    `>(!&t)->` environment mutation.
+  - `uns >ld-> unit`, `uns >mut-> unit`: Environment `ld`/`mut` effect during function application.
+    Type-agnostic environment effects impact mutable values of all types.
+  - `~alloc->`, `~{alloc|hlt}->`, `>e~alloc->`: Explicit [never-parametric] concealed effect(s). An
+    explicitly pure application is written as `~->`.
+- Mutability (`mutability`)
+  - `val v ^m: t m`, `val v ^t ^m: ^t&t m`, `val v ^t ^m: ^&t m`: Parametric mutability. `^m` is a
+    subsidiary mutability parameter, whereas `^t` comes after the type name and is the primary
+    mutability parameter. `^t&t` can be shortened to `^&t`.
+  - `_&t`: Mutability is contextually irrelevant, and therefore both immutable and mutable values
+    are supported.
+  - `&t`: Non-parametric primary mutability. `!&t` indicates a mutation effect on the value, which
+    must therefore be mutable, as in `(a: type) -> ... -> !&t a -> ...`.
+  - `\&t`: Transitive prohibited mutability constraint.
+- Type (`type`)
+  - `type a 'a: a`, `type a 'aa ^a ^m >e: a^ aa m e`: Parametric type.
```

Comment:
It looks like `a^` is a shortening of `a^a`. It's a little weird that shortened syntax differs from `^t&t` to `^&t`, though I see why `a^a` can't be shortened to `^a` using a similar rule (it's ambiguous with the effect parameter syntax).

It makes me wonder if it would be better to just apply the shortening sigil on the right side all the time to keep things consistent.
`^t&t` shortens to `&t^` instead of `^&t`. Perhaps it'd be better to write `^t&t` as `&t^t` and say `&t^t` shortens to `&t^`. It also fits a little better in my head to attach the primary mutability parameter on the right side of the type (almost like a parameter to the type, except without a space to indicate primary mutability).

Reply:
Whoops, `a^` should be `^&a`.